tests/components/caldav/test_calendar.py

```diff
@@ -939,5 +939,8 @@ async def test_get_events_custom_calendars(hass, calendar, get_api_events):
             "summary": "This is a normal event",
             "location": "Hamburg",
             "description": "Surprisingly rainy",
+            "uid": None,
```

Comment:
Why do we update other calendar integration tests?

Reply:
Two things happening here:
1) This PR changes the entity model that any integration can use. These new fields are in use by local calendar, but will also be used by google calendar.
2) The current behavior of the rest API is to return a field (e.g. `description`) even when it is not set. my impression is that some APIs prefer this so that callers don't have to handle multiple conditions of unset and set to null, but I don't have a preference here.  PR https://github.com/home-assistant/core/pull/79598 codifies this behavior more clearly showing the differences between the API return values and the other `as_dict` method used in event payloads.

From your comment, i take it that you'd like #2 to omit `None` fields. Am I understanding that right?